File: examples/ListPeople.cs
// See CSHARP-README.txt for information and build instructions.
// 
using System;
using System.IO;
using Tutorial;

class ListPeople {
  // Iterates though all people in the AddressBook and prints info about them.
  static void Print(AddressBook addressBook) {
    foreach (Person person in addressBook.PersonList) {
      Console.WriteLine("Person ID: {0}", person.Id);
      Console.WriteLine("  Name: {0}", person.Name);
      if (person.HasEmail) {
        Console.WriteLine("  E-mail address: {0}", person.Email);
      }
      
      foreach (Person.Types.PhoneNumber phoneNumber in person.PhoneList) {
        switch (phoneNumber.Type) {
          case Person.Types.PhoneType.MOBILE:
            Console.Write("  Mobile phone #: ");
            break;
          case Person.Types.PhoneType.HOME:
            Console.Write("  Home phone #: ");
            break;
          case Person.Types.PhoneType.WORK:
            Console.Write("  Work phone #: ");
            break;
        }
        Console.WriteLine(phoneNumber.Number);
      }
    }
  }
  
  // Main function:  Reads the entire address book from a file and prints all
  // the information inside.
  public static int Main(string[] args) {
    if (args.Length != 1) {
      Console.Error.WriteLine("Usage:  ListPeople ADDRESS_BOOK_FILE");
      return -1;
    }
    
    // Read the existing address book.
    using (Stream stream = File.OpenRead(args[0])) {
      AddressBook addressBook = AddressBook.ParseFrom(stream);
      Print(addressBook);
    }
    return 0;
  }
}

File: examples/list_people.dart
import 'dart:io';

import 'dart_tutorial/addressbook.pb.dart';
import 'dart_tutorial/addressbook.pbenum.dart';

/// Iterates though all people in the AddressBook and prints info about them.
void printAddressBook(AddressBook addressBook) {
  for (var person in addressBook.people) {
    print('Person ID: ${person.id}');
    print('  Name: ${person.name}');
    if (person.hasEmail()) {
      print('  E-mail address:${person.email}');
    }

    for (var phoneNumber in person.phones) {
      switch (phoneNumber.type) {
        case Person_PhoneType.MOBILE:
          print('   Mobile phone #: ');
          break;
        case Person_PhoneType.HOME:
          print('   Home phone #: ');
          break;
        case Person_PhoneType.WORK:
          print('   Work phone #: ');
          break;
        default:
          print('   Unknown phone #: ');
          break;
      }
      print(phoneNumber.number);
    }
  }
}

/// Reads the entire address book from a file and prints all
/// the information inside.
void main(List<String> arguments) {
  if (arguments.length != 1) {
    print('Usage: list_person ADDRESS_BOOK_FILE');
    exit(-1);
  }

  // Read the existing address book.
  final file = new File(arguments.first);
  final addressBook = new AddressBook.fromBuffer(file.readAsBytesSync());
  printAddressBook(addressBook);
}
